zwad/ad/__init__.py

```diff
@@ -13,7 +15,7 @@
 from zwad.ad.util import run_classifier, fetch_anomalies
 
 
-class ZtfAnomalyDetector:
+class BaseAnomalyDetector(ABC):
```

Comment:
Personally I don't like abc-classes because:
1. They sometimes mess with my own metaclasses.
2. They impose too many restrictions. For example, if we inherit from some abc-class then we have to define all the abstract methods in order to instantiate even a single instance of descendant class. That's not always the intended behavior.

All after all, if you guys are happy with abc, I have no objections.